strawberry/aiohttp/views.py

```diff
@@ -0,0 +1,106 @@
+import json
+from io import BytesIO
+from pathlib import Path
+from typing import Any, Optional, Type
+
+import aiohttp.web
+from strawberry.file_uploads.data import replace_placeholders_with_files
+from strawberry.http import GraphQLHTTPResponse, process_result
+from strawberry.schema import BaseSchema
+from strawberry.types import ExecutionContext, ExecutionResult
+
+
+class GraphQLView(aiohttp.web.View):
+    graphiql = True
+    schema: Optional[BaseSchema] = None
+
+    async def get_root_value(self) -> Any:
+        return None
+
+    async def get_context(self) -> Any:
+        return {"request": self.request}
+
+    async def process_result(self, result: ExecutionResult) -> GraphQLHTTPResponse:
+        return process_result(result)
+
+    async def get(self) -> aiohttp.web.Response:
+        if self.should_render_graphiql:
+            return self.render_graphiql()
+        return aiohttp.web.HTTPNotFound()
+
+    async def post(self) -> aiohttp.web.Response:
+        operation_context = await self.get_execution_context()
+        context = await self.get_context()
+        root_value = await self.get_root_value()
+
+        result = await self.schema.execute(
+            query=operation_context.query,
+            root_value=root_value,
+            variable_values=operation_context.variables,
+            context_value=context,
+            operation_name=operation_context.operation_name,
+        )
+
+        response_data = await self.process_result(result)
+        return aiohttp.web.json_response(response_data)
+
+    async def get_execution_context(self) -> ExecutionContext:
+        try:
+            data = await self.parse_body()
+        except json.JSONDecodeError:
+            raise aiohttp.web.HTTPBadRequest(
+                reason="Unable to parse request body as JSON"
+            )
+
+        try:
+            query = data["query"]
+            variables = data.get("variables")
+            operation_name = data.get("operationName")
```

Comment:
Nitpick: Can you move these out of the `try/except` as they cannot raise a `KeyError`?